src/nvim/option.c

```diff
@@ -598,23 +598,20 @@ void set_init_1(void)
     }
   }
 
-  /*
-   * 'maxmemtot' and 'maxmem' may have to be adjusted for available memory
-   */
+  // 'maxmemtot' and 'maxmem' are ignored in Neovim, but we set them to the
+  // amount of available memory.
```

Comment:
Let's say these options are "readonly". So users can still read the values, but setting the values should be an error.

Also need to note this in vim_diff.txt.